pandas/io/tests/test_excel.py

```diff
@@ -416,48 +314,114 @@ def test_reader_converters(self):
                       3: lambda x: str(x) if x else '',
                       }
 
-        xlsx_path = os.path.join(self.dirpath, 'test_converters.xlsx')
-        xls_path = os.path.join(self.dirpath, 'test_converters.xls')
-
         # should read in correctly and set types of single cells (not array dtypes)
-        for path in (xls_path, xlsx_path):
-            actual = read_excel(path, 'Sheet1', converters=converters)
-            tm.assert_frame_equal(actual, expected)
+        actual = read_excel(pth, 'Sheet1', converters=converters)
+        tm.assert_frame_equal(actual, expected)
 
     def test_reading_all_sheets(self):
         # Test reading all sheetnames by setting sheetname to None,
         # Ensure a dict is returned.
         # See PR #9450
-
-        _skip_if_no_xlrd()
-
-        dfs = read_excel(self.multisheet,sheetname=None)
-        expected_keys = ['Alpha','Beta','Charlie']
-        tm.assert_contains_all(expected_keys,dfs.keys())
+        pth = os.path.join(self.dirpath, 'test_multisheet' + self.ext)
+        dfs = read_excel(pth, sheetname=None)
+        expected_keys = ['Alpha', 'Beta', 'Charlie']
+        tm.assert_contains_all(expected_keys, dfs.keys())
 
     def test_reading_multiple_specific_sheets(self):
         # Test reading specific sheetnames by specifying a mixed list
         # of integers and strings, and confirm that duplicated sheet
         # references (positions/names) are removed properly.
-
         # Ensure a dict is returned
         # See PR #9450
-        _skip_if_no_xlrd()
-
-        #Explicitly request duplicates.  Only the set should be returned.
-        expected_keys = [2,'Charlie','Charlie']
-        dfs = read_excel(self.multisheet,sheetname=expected_keys)
+        pth = os.path.join(self.dirpath, 'test_multisheet' + self.ext)
+        # Explicitly request duplicates. Only the set should be returned.
+        expected_keys = [2, 'Charlie', 'Charlie']
+        dfs = read_excel(pth, sheetname=expected_keys)
         expected_keys = list(set(expected_keys))
-        tm.assert_contains_all(expected_keys,dfs.keys())
+        tm.assert_contains_all(expected_keys, dfs.keys())
         assert len(expected_keys) == len(dfs.keys())
 
+
+class XlrdTests(ReadingTestsBase):
+    """
```

Comment:
I wanted to have the reader agnostic tests in their own class so they can be re-used more conveniently when adding additional readers. I have been trying to do that for adding ODF/ODS (Open Document Format, LibreOffice) support (see also PR #9070). Tests for another reader would inherit from ```ReadingTestsBase```. Since that is not happening yet, I could put all tests from ```XlrdTests``` into ```ReadingTestsBase```. To me it makes sense to have a reader agnostic ```ReadingTestsBase```, but just let me know what you prefer here.